header.cmake

```diff
@@ -71,6 +71,10 @@ MACRO(_SETUP_PROJECT_HEADER)
     STRING(REGEX REPLACE "[^a-zA-Z0-9]" "/" HEADER_DIR "${PROJECT_NAME}")
   ENDIF(DEFINED CUSTOM_HEADER_DIR)
 
+  IF(NOT DEFINED CUSTOM_HEADER_EXTENSION)
```

Comment:
1. This should be documented as other global variables that have an influence on the modules behavior (see `base.cmake`)
2. I suggest renaming the variable `PROJECT_CUSTOM_HEADER_EXTENSION` for consistency with other such options

Reply:
I applied the request modifications.